js/gateio.js

```diff
@@ -1951,27 +1960,52 @@ module.exports = class gateio extends Exchange {
         //         type: 'sell'
         //     }
         //
-        // private
-        //
-        //     {
-        //         "id": "218087755",
-        //         "create_time": "1578958740",
-        //         "create_time_ms": "1578958740122.710000",
-        //         "currency_pair": "BTC_USDT",
-        //         "side": "sell",
-        //         "role": "taker",
-        //         "amount": "0.0004",
-        //         "price": "8112.77",
-        //         "order_id": "8445563839",
-        //         "fee": "0.006490216",
-        //         "fee_currency": "USDT",
-        //         "point_fee": "0",
-        //         "gt_fee": "0"
-        //     }
+        // SPOT rest
+        //  {
+        //      "id":"2876130500",
+        //      "create_time":"1645464610",
+        //      "create_time_ms":"1645464610777.399200",
+        //      "currency_pair":"DOGE_USDT",
+        //      "side":"sell",
+        //      "role":"taker",
+        //      "amount":"10.97",
+        //      "price":"0.137384",
+        //      "order_id":"125924049993",
+        //      "fee":"0.00301420496",
+        //      "fee_currency":"USDT",
+        //      "point_fee":"0","gt_fee":"0"
+        //  }
+        //
+        // Perpetual Swap rest
+        //  {
+        //      "size":-5,
+        //      "order_id":"130264979823",
+        //      "id":26884791,
+        //      "role":"taker",
+        //      "create_time":1645465199.5472,
+        //      "contract":"DOGE_USDT",
+        //      "price":"0.136888"
+        //  }
+        //
+        // Future rest
+        //  {
+        //     "id": 121234231,
+        //     "create_time": 1514764800.123,
+        //     "contract": "BTC_USDT",
+        //     "order_id": "21893289839",
+        //     "size": 100,
+        //     "price": "100.123",
+        //     "role": "taker"
+        //  }]
         //
         const id = this.safeString (trade, 'id');
         let timestamp = this.safeTimestamp (trade, 'time');
-        timestamp = this.safeInteger (trade, 'create_time_ms', timestamp);
+        if (timestamp === undefined) {
+            timestamp = this.safeInteger (trade, 'create_time_ms');
+            if (timestamp === undefined) {
+                timestamp = this.safeIntegerProduct (trade, 'create_time', 1000);
```

Comment:
instead of this code you can write it like this:
```
let timestamp = this.safeTimestamp2 (trade, 'time', 'create_time_ms');
if (timestamp === undefined) {
    timestamp = this.safeInteger (trade, 'create_time_ms');
}
```

it's a bit cleaner

Reply:
sure thing, will fix it now

Reply:
@frosty00 pushed a cleanup which works better in this case and in the style that you showed, lmk what you think